rfc/src/rfcs/0007-global-conditions.md

```diff
@@ -0,0 +1,111 @@
+
+## User Experience
+
+**The output will refer to properties that depend on other properties as "global conditions"**, which is a simpler term.
+The options to enable different global conditions will depend on a case-by-case basis[^enable-options].
+
+The main UI change in this proposal is a new `GLOBAL CONDITIONS` section that **won't be printed if no global conditions have been enabled**.
+This section will only appear in Kani's default output after the `RESULTS` section (used for individual checks) and have the format:
+
+```
+GLOBAL CONDITIONS:
+ - `<name>`: <status> (<reason>)
+ - `<name>`: <status> (<reason>)
+ [...]
+```
+
+where:
+  - `<name>` is the name given to the global condition.
+  - `<status>` is the status determined for the global condition.
+  - `<reason>` is an explanation that depends on the status of the global condition.
+
+For example, let's assume we implement the option requested in [#2299](https://github.com/model-checking/kani/issues/2299).
+A concrete example of this output would be:
+
+```
+GLOBAL CONDITIONS:
+ - `fail_uncoverable`: SUCCESS (all cover statements were satisfied as expected)
+```
+
+A `FAILED` status in any enabled global condition will cause verification to fail.
+In that case, the overall verification result will point out that one or more global conditions failed, as in:
+
+```
+VERIFICATION:- FAILURE (one or more global conditions failed)
+```
+
+This last UI change will also be implemented for the terse output.
+Finally, checks that cause an enabled global condition to fail will be reported using the same interface we use for failed checks[^failed-checks].
+
+**Global conditions which aren't enabled won't appear in the `GLOBAL CONDITIONS` section**.
+Their status will be computed regardless[^status-computation], and we may consider showing this status when the `--verbose` option is passed.
+
+The documentation of global conditions will depend on how they're enabled, which depends on a case-by-case basis.
+However, we may consider adding a new subsection `Global conditions` to the `Reference` section that collects all of them so it's easier for users to consult all of them in one place.
+
+## Detailed Design
+
+The only component to be modified is `kani-driver` since that's where verification results are built and determined.
+But we should consider moving this logic into another crate.
+
+We don't need new dependencies.
+The corner cases will depend on the specific global conditions to be implemented.
+
+## Rationale and alternatives
+
+As mentioned earlier, we're proposing this change to help users understand global conditions and how they're determined.
+In many cases, global conditions empower users to write harnesses which weren't possible to write before.
+As an example, the `#[kani::should_panic]` attribute allowed users to write harnesses expecting panic-related failures.
```

Comment:
the documentation of `should_panic` does not really give a precise meaning to the check, i.e. it does not say if at least one execution must trigger a panic or if all executions must trigger a panic. How can it be modelled in terms of dependent conditions  and statuses ?

Reply:
I don't think we do that for any of the checks we currently produce. Regardless, I think it's well defined in [here](https://model-checking.github.io/kani/reference/attributes.html#kanishould_panic), and if you think otherwise I'm happy to do any editions you consider appropriate.